apps/angular/src/app/features/auth/components/registration-form/registration-form.component.ts

```diff
@@ -0,0 +1,132 @@
+import { ChangeDetectionStrategy, Component, DestroyRef, OnInit, inject } from '@angular/core';
+import { AsyncPipe, CommonModule } from '@angular/common';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatButtonModule } from '@angular/material/button';
+import { FormBuilder, FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { ValidationService } from '@js-camp/angular/core/services/validation.service';
+import { AuthService } from '@js-camp/angular/core/services/auth.service';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { BehaviorSubject, catchError, tap, throwError } from 'rxjs';
+import { HttpErrorResponse } from '@angular/common/http';
+import { HttpErrors } from '@js-camp/angular/core/models/http-errors';
+
+import { CURRENT_AUTH_FORM$ } from '@js-camp/angular/shared/constants/current-auth-form';
+import { CurrentForm } from '@js-camp/angular/shared/constants/current-auth-form-enum';
+import { NavigationService } from '@js-camp/angular/core/services/navigation.service';
+
+import { ErrorComponent } from '../error/error.component';
+
+/** Registration form. */
+type RegistrationForm = {
+
+	/** Email. */
+	readonly email: FormControl<string>;
+
+	/** First name. */
+	readonly firstName: FormControl<string>;
+
+	/** Last name. */
+	readonly lastName: FormControl<string>;
+
+	/** Password. */
+	readonly password: FormControl<string>;
+
+	/** Retyped password. */
+	readonly retypedPassword: FormControl<string>;
+};
+
+/** Registration form component. */
+@Component({
+	selector: 'camp-registration-form',
+	standalone: true,
+	imports: [
+		CommonModule,
+		MatFormFieldModule,
+		MatInputModule,
+		MatButtonModule,
+		ReactiveFormsModule,
+		ErrorComponent,
+		AsyncPipe,
+	],
+	templateUrl: './registration-form.component.html',
+	styleUrl: './registration-form.component.css',
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RegistrationFormComponent implements OnInit {
+	/** Validation service. */
+	protected readonly validationService = inject(ValidationService);
+
+	private readonly authService = inject(AuthService);
+
+	private readonly navigationService = inject(NavigationService);
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	private readonly formBuilder = inject(FormBuilder);
+
+	private readonly currentAuthForm$ = CURRENT_AUTH_FORM$;
+
+	/** Registration form. */
+	protected readonly registrationForm: FormGroup<RegistrationForm> = this.formBuilder.nonNullable.group(
+		{
+			email: ['', [Validators.required, Validators.email]],
```

Comment:
```suggestion
			email: this.formBuilder.control('', [Validators.required, Validators.email]),
```
If you start working with arrays, you may run into a typing problem. Let's use the form creation mechanisms provided by formBuilder
https://wiki.saritasa.rocks/frontend/frameworks/angular/how-tos#reusable-form